### ISSUE TITLE
[ML]: Upgrade prediction model from Logistic Regression to XGBoost and add SHAP explainability

### ISSUE THREAD

**Issue by sonusharma6-dsa** [Arnav-Singh-5080/IPL-Win-Probability-Prediction#362]
### Summary
The current model uses **Logistic Regression** which achieves reasonable accuracy. This issue proposes upgrading to **XGBoost** and adding **SHAP (SHapley Additive exPlanations)** visualizations for model interpretability.

### Motivation
- XGBoost typically achieves 5-10% better accuracy on tabular sports data
- SHAP values show *why* a prediction was made — critical for user trust in sports analytics
- The `requirements.txt` already includes `xgboost`, so the dependency is ready

### Proposed Changes

**1. Model Upgrade (`application.py`)**
\\\python
from xgboost import XGBClassifier

xgb_model = XGBClassifier(
    n_estimators=200,
    max_depth=6,
    learning_rate=0.05,
    use_label_encoder=False,
    eval_metric='logloss'
)
\\\

**2. Model Comparison Report**
- Add a model benchmarking section showing LogReg vs XGBoost accuracy
- Include cross-validation scores (5-fold CV)

**3. SHAP Explainability Panel**
\\\python
import shap
explainer = shap.TreeExplainer(xgb_model)
shap_values = explainer.shap_values(X_input)
shap.summary_plot(shap_values, X_input)
\\\

**4. New Model File**
- Save as `xgb_model.pkl` alongside existing `logistic_model.pkl`
- Keep backward compatibility with old model

### Acceptance Criteria
- [ ] XGBoost model trained and saved as `xgb_model.pkl`
- [ ] Accuracy comparison table shown in UI
- [ ] SHAP bar chart displayed after each prediction
- [ ] 5-fold cross-validation score reported
- [ ] Existing LogReg model still accessible as fallback
- [ ] README model metrics section updated

### Difficulty
🔴 **Hard** — requires ML expertise with XGBoost and SHAP

---
*Raised as part of GSSoC 26 contribution to CricScope.*

**Comment by sonusharma6-dsa** [comment]
Hi @Arnav-Singh-5080 👋

I would like to work on this issue as part of **GSSoC '26**.

I have reviewed the current `logistic_model.pkl` pipeline and noticed that `xgboost` is already listed in `requirements.txt`, which means the upgrade can be done without adding new dependencies.

**My plan:**
- Train an XGBoost classifier on the same dataset and save as `xgb_model.pkl`
- Add SHAP TreeExplainer for post-prediction feature importance visualization
- Display a side-by-side accuracy comparison (LogReg vs XGBoost) in the UI
- Include 5-fold cross-validation scores
- Keep the existing LogReg model as a fallback

Could you please assign this issue to me? 🙏

**Comment by Vectorique2506** [comment]
Hello! @Arnav-Singh-5080
I have done many projects using XGBOOST model, I believe i have the necessary expertise and experience in managing and solving this issue pls assign me this issue 

Thanks a ton !

**Comment by Abhi-0111** [comment]
Hi @Arnav-Singh-5080 Could you please assign this issue to me under gssoc 26'

**Comment by PranavAgarkar07** [comment]
/claim

Will implement: train an XGBClassifier with 5-fold CV, generate SHAP tree explainer visualizations (summary + bar plots), add a model comparison table between LogReg and XGBoost, save as `xgb_model.pkl`, and keep the existing LogReg fallback. Plotly charts will match the dark gold theme.

**Comment by Shreya5473** [comment]
Hi, I am a GSSoC 2026 contributor and would like to work on this. I have worked on this before and I would love to complete this issue.

**Comment by ujala100** [comment]
Hi @Arnav-Singh-5080

I’m interested in contributing to this project and would love to work on this issue. I have experience working on open-source contributions and would be happy to implement the required changes.

Could you please assign this issue to me? Thanks
